.dockerignore

```diff
@@ -0,0 +1,42 @@
+# Reminders:
+# * Matching rules are different to `.gitignore`
+# * A pattern without '**' matches in the project's root directory only
+# * Leading and trailing '/' are discarded (it is not possible to
+#   distinguish between files and directories)
+# * More details: https://github.com/balena-io-modules/dockerignore
+
+# development and testing tools or IDEs
+**/*.log
+**/*.pid
+**/*.seed
+.idea
+.lock-wscript
+.nvmrc
+.nyc_output
+.vscode
+coverage
+lib-cov
+logs
+pids
+
+# OS cache files
+**/.DS_Store
```

Comment:
I have added a commit to review these patterns. A difference between `.gitignore` and `.dockerignore` is that a pattern like `.DS_Store` applies to subdirectories as well in `.gitignore`, but only to the project root directory in `.dockerignore`. So I rewrote some such `.dockerignore` patterns as `**/.DS_Store`.
I have also added comments at the top of the the files to highlight other differences between them.

Reply:
I just noticed that I force pushed over these changes! Do you still have the .dockerignore and .gitignore locally?

Reply:
Nevermind, I found the orphaned commit and reapplied it!

Reply:
I had myself removed my commit to test whether it was the cause of the build failures. It wasn't -- the build was failing with or without that commit, and I had not understood why. Let's see if the build passes now...

Reply:
Still seeing the build failures, but I can't figure out from the logs what's wrong. They build fine locally?

Reply:
Yes, the build works on my laptop as well. By the way, As of Docker Desktop v3.2.1 (macOS), Docker Engine v20.10.5, the default builder is buildkit, and the output of `docker build` is quite different than it used to be. So now my machine is quite different from the CI. I don't particularly like that it is different, but I get to see what many CLI end users will be experiencing as well.

I don't know what's wrong, but we can merge the PR until we figure it out... If in doubt, perhaps you could create a separate PR where you push the last known-to-succeed commit, if you still have it - or something really basic, like a single line Dockerfile with just a FROM instruction, which might succeed or fail... Then you could ping the CI team (Giovanni...) and ask - "what's changed in the CI? This build used to pass!"

Reply:
I pushed a change to remove the base image pinning, which is one of the changes I made early in the week.

Reply:
Found the issue, some of the images don't exist with that date tag. I'll move the base image definitions to `.resinci.yml` so it can be unique per arch/distro and make it easier to automate updating in the future.

Reply:
The latest v12 available for i386-debian-node is 12.16.3, but the rest are at various releases of 12.19.1.
Do you think we should align them all on 12.16.3 (if possible) or whatever the latest v12 is for the respective platforms?

Reply:
It is already the case that a single CLI release / version uses different versions of Node.js bundled in the different installers: GUI installers for Windows and macOS, and the standalone zip packages. I don't readily see the benefit of having all Docker images of a CLI release use the same version of Node.js (if was even possible), but also it's not a big deal either way. So it's really up to you... If it was me making the choice, I would probably have each Docker image use the latest available base image image version for that arch-distro combination.

Reply:
Having each Docker image use the latest available base image for that arch-distro combination is also probably slightly easier to automate, as it avoids having to find the latest common version -- which may not even exist for all images. And if one of the base images was lagging behind the other for any reason -- e.g. one image in Node v10 and the other images in Node v14 -- we would not want to penalise all other images because of that one. So I am actually starting to see a reason to have each image use the latest available base image for its arch-distro combination.

Reply:
Agreed, that's what I was thinking as well. Nice to see the builds working now!